File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    count = {}
    
    for i in clothes:
        if i[1] in count:
            count[i[1]] += 1
        else:
            count[i[1]] = 1
            
    for j in count:
        answer *= count[j] + 1
    
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        Map<String, Integer> clothesCounts = new HashMap<>();
        
        for (String[] items : clothes) {
            String clothesType = items[1];
            clothesCounts.put(
                clothesType,
                clothesCounts.getOrDefault(clothesType, 0) + 1
            );
        }
        
        for (int count : clothesCounts.values()) {
            answer *= count + 1;
        }
        
        answer--;
        return answer;
    }
}